draft-parecki-oauth-client-id-metadata-document.md

```diff
@@ -102,12 +104,37 @@ The client metadata document URL is a JSON document containing the metadata
 of the client. The client metadata values are the values defined in
 OAuth Dynamic Client Registration ({{RFC7591}}) section 2.
 
+The client metadata document MUST contain a `client_id` property whose value
+MUST compare and match the URL of the document using simple string comparison as
+defined in [RFC3986] Section 6.2.1.
+
+The client metadata document MAY define additional properties in the response.
+The client metadata document MAY also be served with more specific content types
+as long as the response is JSON and conforms to application/<AS-defined>+json.
+
+Other specifications MAY place additional restrictions on the contents of the
+client metadata document accepted by authorization servers implementing their
+specification, for instance, preventing the registration of confidential clients
+by requiring the `token_endpoint_auth_method` property be set to `"none"`.
```

Comment:
I'm not sure if I have the language here correct. But basically the idea is that authorization servers may reject specific client types or mandate that certain standard properties must be present and have a given value.